frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,368 @@
+          {showLabelSelectors && (
+            <NameValueEditor
+              nameValuePairs={nameValuePairs}
+              valueString="Selector"
+              nameString="Label"
+              addString="Add Value"
+              readOnly={false}
+              allowSorting={false}
+              updateParentData={this.handleNameValuePairs}
+            />
+          )}
+          <p className="form-text text-muted" id="label-selector-help">
+            Use label selectors to define how storage is created.
+          </p>
+        </div>
+      </div>
+
+    );
+  }
+}
+
+class CreatePvcPage extends React.Component<CreatePvcPageProps, CreatePvcPageState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: '',
+      inProgress: false,
+      title: 'Create Storage',
+      pvcObj: {},
+    };
+  }
+
+  onChange = pvcObj => {
+    this.setState({ pvcObj });
+  };
+
+  validateForm = () => {
+    if (!_.get(this.state.pvcObj, 'metadata.name')) {
+      this.setState({ error: 'Persistent volume claim name must be set.' });
+      return false;
+    }
+    return true;
+  };
+
+  save = e => {
+    if (!this.validateForm()) {
+      return;
+    }
+    e.preventDefault();
+    this.setState({ inProgress: true });
+    k8sCreate(this.props.kindObj, this.state.pvcObj).then(
+      resource => {
+        this.setState({ inProgress: false });
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      },
+      err => this.setState({ error: err.message, inProgress: false })
+    );
+  };
+
+  render() {
+    const { title, error, inProgress } = this.state;
+    const { namespace, kindObj } = this.props;
+    return (
+      <div className="co-m-pane__body">
+        <Helmet>
+          <title>{title}</title>
+        </Helmet>
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <form className="co-m-pane__body-group " onSubmit={this.save}>
+          <CreatePvcForm onChange={this.onChange} isCreate={true} namespace={namespace} kindObj={kindObj} />
+          <ButtonBar errorMessage={error} inProgress={inProgress}>
+            <button type="submit" className="btn btn-primary" id="save-changes">
+              Create
+            </button>
+            <button type="button" className="btn btn-default" onClick={history.goBack}>
+              Cancel
+            </button>
+          </ButtonBar>
+        </form>
+      </div>
+    );
+  }
+}
+
+export const CreatePvc_ = ({ kindObj, kindsInFlight, match: { params } }) => {
+  if (!kindObj && kindsInFlight) {
+    return <LoadingBox />;
+  }
+  return <CreatePvcPage isCreate={true} namespace={params.ns} kindObj={kindObj} />;
+};
+
+export const CreatePvc = connectToPlural(CreatePvc_);
+
+export type StorageClassDropdownProps = {
+  namespace: string;
+  selectedKey: string;
+  required: boolean;
+  onChange: Function;
+  id: string;
+  name: string;
+  placeholder?: string;
+};
+
+export type CreatePvcFormProps = {
+  kindObj: K8sKind;
```

Comment:
You shouldn't need to pass `kindObj` in. You know it's a PVC kind and can just use the model.